vulnerabilities/views.py

```diff
@@ -96,17 +104,34 @@ class VulnerabilitySearch(ListView):
     ordering = ["vulnerability_id"]
     paginate_by = PAGE_SIZE
 
+    def get_paginate_by(self, queryset):
+        page_size = self.request.GET.get("page_size", "")
+        return int(page_size) if page_size.isdigit() else self.paginate_by
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         request_query = self.request.GET
         context["vulnerability_search_form"] = VulnerabilitySearchForm(request_query)
         context["search"] = request_query.get("search")
+        context["page_size"] = self.get_paginate_by(self.get_queryset())
         return context
 
-    def get_queryset(self, query=None):
-        query = query or self.request.GET.get("search") or ""
+    def get_queryset(self):
+        query = self.request.GET.get("search") or ""
         return self.model.objects.search(query=query).with_package_counts()
```

Comment:
That logic and everything related to search/ordering/filering should be handle by the *SearchForm and not by the view itself.

Reply:
Added the search/ordering/filtering functionality in `BaseSearchForm` , `PackageSearchForm` , `VulnerabilitySearchForm`in forms.py.